pkg/istio-agent/agent.go

```diff
@@ -575,7 +577,26 @@ func (a *Agent) Check() (err error) {
 
 func (a *Agent) GetDNSTable() *dnsProto.NameTable {
 	if a.localDNSServer != nil {
-		return a.localDNSServer.NameTable()
+		nt := a.localDNSServer.NameTable()
+		nt = proto.Clone(nt).(*dnsProto.NameTable)
+		a.localDNSServer.BuildAlternateHosts(nt, func(althosts map[string]struct{}, ipv4 []net.IP, ipv6 []net.IP, _ []string) {
```

Comment:
why can't we use `buildDNSAnswers` here? instead of recreating it?

Reply:
because the code is tightly couple to lookup table and populates a and aaaa records - we do not need to have that information here though.

May be possible to refactor both to reuse this piece of code.
Is that what you are suggestinb?